### PR TITLE
Nodejs config options

### PR DESCRIPTION
**Pull request by thijsc**
I've added the Node.js config options I know to work with basic descriptions. Leaving #371 open since this still needs more attention.

I could not get Markdown rendering to work and I do not understand why. Got this error:

```NoMethodError at /nodejs/configuration/options.html
undefined method `link_to' for nil:NilClass

Ruby	/Users/thijs/.rbenv/versions/2.4.5/lib/ruby/gems/2.4.0/gems/middleman-core-4.2.1/lib/middleman-core/renderers/redcarpet.rb: in link, line 89
Web	GET localhost/nodejs/configuration/options.html
```

I've added a temp fix so I could see the page in 79b3b2f. @tombruijn do you maybe have any idea what could be causing this error? I've looked quite hard but could not find a reason.

### CODE REVIEW

File: config.rb [hunk 1]
```diff
@@ -126,7 +126,7 @@ def edit_link
 
   def markdown(content)
     # https://github.com/hashicorp/middleman-hashicorp/blob/14c705614b2f97b5a78903f17b904f767c1fdbe2/lib/middleman-hashicorp/redcarpet.rb
-    Redcarpet::Markdown.new(AppsignalMarkdown, AppsignalMarkdown::OPTIONS).render(content)
+    Redcarpet::Markdown.new(Redcarpet::Render::HTML, AppsignalMarkdown::OPTIONS).render(content)
```
Comment (tombruijn) [inline config.rb:129]: Please undo this change so the markdown works the same everywhere. Now the special markdown syntax doesn't work on the config options page.

File: data/config_options.yml [hunk 2]
```diff
@@ -192,54 +214,37 @@ options:
...
-        If you're affected by our [DNS timeouts](/support/known-issues.html#dns-timeouts), try setting a DNS server manually using this option that doesn't use more than __4__ dots in the server name.
+      If you're affected by our [DNS timeouts](/support/known-issues.html#dns-timeouts), try setting a DNS server manually using this option that doesn't use more than __4__ dots in the server name.
 
-        - Acceptable values: `8.8.8.8`, `my.custom.local.server`.
-        - Not acceptable values: `foo`, `my.awesome.custom.local.dns.server`.
+      - Acceptable values: `8.8.8.8`, `my.custom.local.server`.
+      - Not acceptable values: `foo`, `my.awesome.custom.local.dns.server`.
 
-        If the DNS server cannot be reached the agent will fall back on the host's DNS configuration and output a message in the `appsignal.log` file: `A problem occurred while setting DNS servers`.
+      If the DNS server cannot be reached the agent will fall back on the host's DNS configuration and output a message in the `appsignal.log` file: `A problem occurred while setting DNS servers`.
```
Comment (tombruijn) [inline data/config_options.yml:247]: You've removed the Ruby description above and indented this description on to the left so it applies to all languages, but only has Elixir examples. Please add a description per language as before.

**Review by tombruijn — CHANGES_REQUESTED** [review]

**Comment by tombruijn** [comment]
> I've added a temp fix so I could see the page in 79b3b2f. @tombruijn do you maybe have any idea what could be causing this error? I've looked quite hard but could not find a reason.

From what I can tell you, after middleman starts, you first need to visit a page that's not the config page, then visit the config page and it renders correctly.

File: config.rb (hunk 1 above)
Comment (thijsc) [reply config.rb:129]: I mentioned this as being a temp fix, I had no intention to merge this.

**Comment by thijsc** [comment]
> From what I can tell you, after middleman starts, you first need to visit a page that's not the config page, then visit the config page and it renders correctly.

Ah so this is a known issue?

**Comment by tombruijn** [comment]
> Ah so this is a known issue?

Not that I know. I just noticed the other day.

File: data/config_options.yml [hunk 3]
```diff
@@ -132,6 +144,9 @@ options:
       since: 0.1.0
       type: string
       default_value: null
+    nodejs:
```
Comment (tombruijn) [inline data/config_options.yml:147]: All the Node.js config options need to have a camelcase version instead of the underscored versions. You can add a `config_key` override for every Node.js config option YAML block.

```suggestion
    nodejs:
      config_key: "pushApiKey"
```

**Comment by thijsc** [comment]
Processed feedback and removed the temporary fix.

File: data/config_options.yml [hunk 4]
```diff
@@ -117,10 +127,14 @@ options:
     elixir:
       type: atom
       default_value: null
-      description: |
-        Name of your application as it should be displayed on AppSignal.com.
+    nodejs:
+      config_key: appName
```
Comment (tombruijn) [inline data/config_options.yml:131]:
```suggestion
```

It's the same as other integrations with `name`:
https://github.com/appsignal/appsignal-nodejs/blob/a75027dacc3ba1920e579218350f64c09a606d1b/packages/nodejs/src/config/configmap.ts#L4

File: data/config_options.yml [hunk 5]
```diff
@@ -103,6 +107,12 @@ options:
         ```
 
         -> **Note**: Changing the [name](#option-name) or environment of an existing app will create a new app on AppSignal.com.
+    nodejs:
+      config_key: appEnv
```
Comment (tombruijn) [inline data/config_options.yml:111]:
```suggestion
      config_key: environment
```

Should be `environment` according to this map: https://github.com/appsignal/appsignal-nodejs/blob/a75027dacc3ba1920e579218350f64c09a606d1b/packages/nodejs/src/config/configmap.ts#L5

File: data/config_options.yml [hunk 6]
```diff
@@ -132,6 +146,10 @@ options:
       since: 0.1.0
       type: string
       default_value: null
+    nodejs:
+      config_key: pushApiKey
```
Comment (tombruijn) [inline data/config_options.yml:150]: My bad, this is actually `apiKey` according to this: https://github.com/appsignal/appsignal-nodejs/blob/a75027dacc3ba1920e579218350f64c09a606d1b/packages/nodejs/src/config/configmap.ts#L3

```suggestion
      config_key: apiKey
```

**Review by tombruijn — COMMENTED** [review]
Fixed some stuff with some descriptions in an added commit..

Can you double check my suggestions for the key names?

**Comment by thijsc** [comment]
Thanks for checking @tombruijn
